src/components/Organization/ViewOrganization/index.js

```diff
@@ -15,33 +15,104 @@ import { clearOrgData, getOrgData } from "../../../store/actions";
 
 const ViewOrganization = () => {
   const { handle } = useParams();
+  const [people, setPeople] = useState([]);
+  const [orgFollowed, setOrgFollowed] = useState([]);
   const firebase = useFirebase();
   const dispatch = useDispatch();
   const firestore = useFirestore();
+  const db = firebase.firestore();
+  const profileData = useSelector(({ firebase: { profile } }) => profile);
 
   const [imageLoading, setImageLoading] = useState(true);
 
+  useEffect(() => {
+    const unsubscribe = db
+      .collection("cl_org_general")
+      .doc(handle)
+      .onSnapshot((snap) => {
+        // const data = snap.docs.map((doc) => doc.data());
+        const data = snap.data();
+        setPeople(data.followers);
+      });
+
+    return () => unsubscribe();
+  }, []);
+
+  useEffect(() => {
+    const unsubscribe = db
+      .collection("cl_user")
+      .doc(profileData.uid)
+      .onSnapshot((snap) => {
+        // const data = snap.docs.map((doc) => doc.data());
+        const data = snap.data();
+        setOrgFollowed(data.orgFollowed);
+      });
+
+    return () => unsubscribe();
+  }, []);
+
+  const addValue = (value) => {
+    if (people && people.includes(value)) {
+      console.log("already followed");
+    } else if (people) {
+      const arr = [...people];
+      arr.push(value);
+      db.collection("cl_org_general").doc(handle).update({
+        followers: arr,
+      });
+      var arr2 = [];
+      if (orgFollowed) arr2 = [...orgFollowed];
+
+      arr2.push(handle);
+      db.collection("cl_user").doc(profileData.uid).update({
+        orgFollowed: arr2,
+      });
+    } else {
+      db.collection("cl_org_general")
+        .doc(profileData.uid)
+        .update({
+          orgFollowed: [handle],
+        });
+    }
+  };
```

Comment:
These database actions are not related to the JSX view, right? Please separate the actions and views into two components and import the actions in the view file. ([Separation of concern](https://nalexn.github.io/separation-of-concerns/)). You can move these actions to `/src/store/actions` directory.

Reply:
okay moving those functions  into store/actions directory

Reply:
moved all the functions which are not related to JSX to ```store/actions/orgActions.js```. @Niweera